src/serializer/ResidualHeapVisitor.js

```diff
@@ -409,9 +413,18 @@ export class ResidualHeapVisitor {
       case "Number":
       case "String":
       case "Boolean":
-      case "ReactElement":
       case "ArrayBuffer":
         return;
+      case "ReactElement":
+        // if we visiting during an additional function, we can check to see if
+        // we can hoist React elements to the main scope
+        let targetScope = this.scope.parent;
```

Comment:
`this.scope.parent` won't always be the main scope, try making some tests with nested scopes that you want to hoist out of. You may want to use `commonScope` (or maybe `commonScope.parent`